src/browser/util/url.js

```diff
@@ -87,6 +93,19 @@ define(function() {
       return !this.urlValidate.validity.valid;
     },
 
+    isAboutURL: function(url) {
+      return url.startsWith('about:');
+    },
+
+    isPrivileged: function urlHelper_isPrivilegedURI(uri) {
+      // FIXME: not safe. White list?
+      return uri && uri.startsWith(this.getBaseURI() + 'src/about/');
+    },
+
+    getManifestURL: function urlHelper_getManifestURL() {
+      return this.getBaseURI() + 'manifest.webapp';
+    },
+
```

Comment:
here also, that would be cleaner to resolve 'src/about/' and 'manifest.webapp' against getBaseURI() using window.URL().